lonestar/analytics/distributed/tc/tc.cpp

```diff
@@ -61,41 +62,69 @@ template <bool async>
 struct TC {
   Graph* graph;
   using DGAccumulatorTy = galois::DGAccumulator<uint64_t>;
-  DGAccumulatorTy& num_triangles;
+  DGAccumulatorTy& numTriangles;
 
-  TC(Graph* _graph, DGAccumulatorTy& _num_triangles)
-      : graph(_graph), num_triangles(_num_triangles) {}
+  TC(Graph* _graph, DGAccumulatorTy& _numTriangles)
+      : graph(_graph), numTriangles(_numTriangles) {}
 
   // use the below line once CPU code is added
-  // void static go(Graph& _graph) {
-  void static go() {
+  void static go(Graph& _graph) {
     unsigned _num_iterations = 0;
-    DGAccumulatorTy num_triangles;
+    DGAccumulatorTy numTriangles;
     syncSubstrate->set_num_round(_num_iterations);
-    num_triangles.reset();
+    numTriangles.reset();
+    const auto& allMasterNodes = _graph.masterNodesRange();
 
 #ifdef GALOIS_ENABLE_GPU
-    if (personality == GPU_CUDA) {
+    if (personality == GPU_CUDA) { ///< GPU TC.
       std::string impl_str(syncSubstrate->get_run_identifier("TC"));
       galois::StatTimer StatTimer_cuda(impl_str.c_str(), regionname);
       StatTimer_cuda.start();
       uint64_t num_local_triangles = 0;
       TC_masterNodes_cuda(num_local_triangles, cuda_ctx);
-      num_triangles += num_local_triangles;
+      numTriangles += num_local_triangles;
       StatTimer_cuda.stop();
+    } else { ///< CPU TC.
+#endif
+      galois::do_all(
+          galois::iterate(allMasterNodes), TC(&_graph, numTriangles),
+          galois::steal(),
+          galois::loopname(syncSubstrate->get_run_identifier("TC").c_str()));
+#ifdef GALOIS_ENABLE_GPU
     }
 #endif
 
-    uint64_t total_triangles = num_triangles.reduce();
+    uint64_t total_triangles = numTriangles.reduce();
     if (galois::runtime::getSystemNetworkInterface().ID == 0) {
       galois::gPrint("Total number of triangles ", total_triangles, "\n");
     }
   }
+
+  void operator()(GNode p1) const {
+    size_t numTriangles_local = 0;
+    for (auto it_p1_1 : graph->edges(p1)) {
```

Comment:
minor comment here: the variable names in this loop are quite hard to understand. Being more descriptive with them instead of using p<number> would make it (1) easier to distinguish them among each other (2) easier to understand what they're used for.

Reply:
Thanks. Maybe v/vv/w used for shared tc cpu would be better than the current p1/p2/p3. I will update it soon.